### ISSUE TITLE
Crossover Debugging Issue

### ISSUE THREAD

**Issue by trav-d13** [kissinger99/MYGA#6]
@kissinger99 I'm responding to the Crossover Debugging Issue here. 
As a note, when mentioning an issue in a Pull Request, don't merge the branch into main before the issue is resolved. 
Rather leave the Pull Request open and we can discuss the issue there. Alternatively Create a new Issue (such as this one) and we can discuss it in the Issues section. 

With regard to the debugging error, it stems from the below code: 
```python
def crossover(parent_1: Individual, parent_2: Individual, population: Population, method: str):
    prob = random.uniform(0,1)

    child_1 = parent_1
    child_2 = parent_2

    if prob <= crossover_probability:
        match method:
            case "Single":
                child_1, child_2 = one_point_crossover(parent_1, parent_2, Population)
            case "Double":
                child_1, child_2 = two_point_crossover(parent_1, parent_2, Population)
    return child_1, child_2
```

Specifically, these lines:
```python
 child_1, child_2 = one_point_crossover(parent_1, parent_2, Population)
```
```python
 child_1, child_2 = two_point_crossover(parent_1, parent_2, Population)
```
You are passing the class `Population` to the crossover methods, instead of the object `population`. The class forms the template for the objects. The objects hold all of the info in this case, especially the `individuals` list. 
To solve it, change the **Population** to **population**. 

Let me know if that helped? 
All the best!!

**Comment by kissinger99** [comment]
It now gives me the error in line 28. --> type object 'Population' has no attribute 'individuals' (why????)

**Comment by trav-d13** [comment]
This is as line 31 in your `Main.py` file currently uses the `Population` instead of `population`. 
Here is the current version: 
``` Python
child_1, child_2 = Crossover.crossover(parent_1, parent_2, Population, "Single")
```

If you correct it to: 
``` Python
child_1, child_2 = Crossover.crossover(parent_1, parent_2, population, "Single")
```
It should work as expected. Let me know if that helped?

**Comment by trav-d13** [comment]
I hear where you're coming from. 
I will happily break it down in the next session. 
Let me know when you're available this week?
